### PR TITLE
Add introspection service

### PR DESCRIPTION
**Pull request by gildub**
Requests for introspection and rules
No Models yet
Note: Requires fog-core => 1.37.0 (Mostly for Mock tests)

### CODE REVIEW

File: fog-openstack.gemspec [hunk 1]
```diff
@@ -31,4 +31,6 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'rubocop'
   spec.add_development_dependency 'rubyzip', '~> 0.9.9'
   spec.add_development_dependency 'shindo',  '~> 0.3'
+  spec.add_development_dependency 'pry'
```
Comment (Ladas) [inline fog-openstack.gemspec]: Hm, lets probably remove this, I haven't seen debugging tools added like this usually. You could just put them to your local gemfile

**Comment by Ladas** [comment]
@gildub looks good in overall, please fix the tests, hound issues and the gemfile nit.

Also could you add some doc under https://github.com/fog/fog-openstack/tree/master/lib/fog/openstack/docs

**Comment by seanhandley** [comment]
This is really promising. Thanks!

Could we use `Fog::Introspection::OpenStack` rather than `Fog::Introspection[:openstack]` ?

**Comment by gildub** [comment]
@ladas, documentation and example added.
@seanhandley, used `Fog::Introspection::OpenStack`

Thanks for following-up. I'm working on the introspections models now, but would rather add along another PR. The reason is this one is quite big (sorry about the coveralls/houndci mess) but at same time is contained (All the requests are in and working).

**Comment by seanhandley** [comment]
Let us know when you're ready for merge @gildub - 1,000+ lines of code will take us a while to review.

**Comment by gildub** [comment]
@seanhandley, thanks, I'm ready for merge. Yes it's a chunk meanwhile this code is mostly green because it adds introspection files and very little of existing code is impacted.

**Comment by Ladas** [comment]
@gildub looks good to me, follows the patterns we have here ,so :+1: Nice docs btw. :-)

**Comment by seanhandley** [comment]
Nice one @gildub 👍

**Pull request merged**